web/src/beta/components/Resizable/index.tsx

```diff
@@ -33,24 +41,50 @@ const Resizable: React.FC<Props> = ({
   const BottomGutter = showBottomGutter ? <HorizontalGutter {...gutterProps} /> : null;
   const LeftGutter = showLeftGutter ? <VerticalGutter {...gutterProps} /> : null;
 
+  useEffect(() => {
+    if (size <= initialSize / 2) {
+      setMinimized(true);
+    } else {
+      setMinimized(false);
+    }
+  }, [direction, initialSize, size]);
+
+  const handleShowOriginalPanel = useCallback(() => {
+    setMinimized(false);
+    onInitializeSize();
+  }, [onInitializeSize]);
   return (
-    <StyledResizable direction={direction} size={size}>
-      {TopGutter}
-      {LeftGutter}
-      <Wrapper>{children}</Wrapper>
-      {RightGutter}
-      {BottomGutter}
-    </StyledResizable>
+    <>
+      {minimized ? (
+        <MinimizedWrapper direction={direction} onClick={handleShowOriginalPanel}>
+          <Icon icon={gutter == "end" ? "arrowRight" : "arrowLeft"} />
+        </MinimizedWrapper>
+      ) : (
+        <StyledResizable direction={direction} size={size} minSize={minSize}>
+          {TopGutter}
+          {LeftGutter}
+          <Wrapper>{children}</Wrapper>
+          {RightGutter}
+          {BottomGutter}
+        </StyledResizable>
+      )}
+    </>
   );
 };
 
-const StyledResizable = styled.div<Pick<Props, "direction" | "size">>`
+const StyledResizable = styled.div<Pick<Props, "direction" | "size" | "minSize">>`
   display: flex;
   align-items: stretch;
   flex-direction: ${({ direction }) => (direction === "vertical" ? "row" : "column")};
   width: ${({ direction, size }) => (direction === "horizontal" ? null : `${size}px`)};
   height: ${({ direction, size }) => (direction === "vertical" ? null : `${size}px`)};
   flex-shrink: 0;
+  resize: both;
+  min-width: ${({ direction, minSize }) =>
+    direction === "horizontal" && minSize ? `${minSize}px` : null};
+  min-height: ${({ direction, minSize }) =>
+    direction === "vertical" && minSize ? `${minSize}px` : null};
+  transition: transform 0.3s ease-in-out;
```

Comment:
I guess this element does not have transform property, so it would be unnecessary.

Reply:
yes sure 👍

Reply:
You can delete [this __line__](https://github.com/reearth/reearth/pull/572/files#diff-c3ec5584f7971867cb38d9a4043cad0e2895c1dddbc2a63b78812b68b8cd482dR80):pray:

```
transition: ease-in-out;
```

This means transition works 0s so the easing (`ease-in-out`) does not change anything.

Reply:
done